pkg/cmd/cli/cmd/rollback.go

```diff
@@ -38,151 +43,267 @@ a human-readable representation of the updated deployment configuration instead
+		deployment, err := o.findTargetDeployment(r, o.DesiredVersion)
+		if err != nil {
+			return err
+		}
+		target = deployment
+	}
+	if target == nil {
+		return fmt.Errorf("%s is not a valid deployment or deploymentconfig", o.TargetName)
+	}
+
+	// Set up the rollback and generate a new rolled back config.
+	rollback := &deployapi.DeploymentConfigRollback{
+		Spec: deployapi.DeploymentConfigRollbackSpec{
+			From: kapi.ObjectReference{
+				Name: target.Name,
+			},
+			IncludeTemplate:        true,
+			IncludeTriggers:        o.IncludeTriggers,
+			IncludeStrategy:        o.IncludeStrategy,
+			IncludeReplicationMeta: o.IncludeScalingSettings,
+		},
+	}
+	newConfig, err := o.oc.DeploymentConfigs(o.Namespace).Rollback(rollback)
+	if err != nil {
+		return err
+	}
+
+	// If this is a dry run, print and exit.
+	if o.DryRun {
+		describer := describe.NewDeploymentConfigDescriberForConfig(o.oc, o.kc, newConfig)
+		description, err := describer.Describe(newConfig.Namespace, newConfig.Name)
+		if err != nil {
+			return err
+		}
+		o.out.Write([]byte(description))
+		return nil
+	}
+
+	// If an output format is specified, print and exit.
+	if len(o.Format) > 0 {
+		printer, _, err := kubectl.GetPrinter(o.Format, o.Template)
+		if err != nil {
+			return err
+		}
+		versionedPrinter := kubectl.NewVersionedPrinter(printer, kapi.Scheme, latest.Version)
+		versionedPrinter.PrintObj(newConfig, o.out)
+		return nil
+	}
+
+	// Perform a real rollback.
+	rolledback, err := o.oc.DeploymentConfigs(newConfig.Namespace).Update(newConfig)
 	if err != nil {
 		return err
 	}
-	versionedPrinter := kubectl.NewVersionedPrinter(printer, kapi.Scheme, latest.Version)
-	versionedPrinter.PrintObj(config, out)
+
+	// Print warnings about any image triggers disabled during the rollback.
+	fmt.Fprintf(o.out, "#%d rolled back to %s\n", rolledback.LatestVersion, rollback.Spec.From.Name)
+	for _, trigger := range rolledback.Triggers {
+		disabled := []string{}
+		if trigger.Type == deployapi.DeploymentTriggerOnImageChange && !trigger.ImageChangeParams.Automatic {
+			disabled = append(disabled, trigger.ImageChangeParams.From.Name)
+		}
+		if len(disabled) > 0 {
+			reenable := fmt.Sprintf("oc deploy %s --enable-triggers", rolledback.Name)
+			fmt.Fprintf(o.out, "Warning: the following images triggers were disabled: %s\n  You can re-enable them with: %s\n", strings.Join(disabled, ","), reenable)
+		}
+	}
+
 	return nil
 }
 
-// Update persists the given DeploymentConfig.
-func (r *helper) Update(config *deployapi.DeploymentConfig) (*deployapi.DeploymentConfig, error) {
-	return r.updateConfig(config.Namespace, config)
+// findResource tries to find a deployment or deploymentconfig named
+// targetName using a resource.Builder. For compatibility, if the resource
+// name is unprefixed, treat it as an rc first and a dc second.
+func (o *RollbackOptions) findResource(targetName string) (runtime.Object, error) {
+	candidates := []string{}
+	if strings.Index(targetName, "/") == -1 {
```

Comment:
> Are we going to do this? @smarterclayton

This is required for backward compatibility.  In v3, we didn't require a resource prefix, but now that we accept two different things we need to allow it so that a user can express exactly what they mean.